### PR TITLE
Handled the case in which child span starts before and ends after parent span.

### LINKED ISSUE #1779: [Bug]: Prevent Critical Path code from throwing exceptions
### What happened?

Critical Path code occasionally throws Range exception here: https://github.com/jaegertracing/jaeger-ui/blob/17eb2916047f7c3d1f5691957e29bbf6784b2e22/packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/sanitizeOverFlowingChildren.tsx#L87

### Steps to reproduce

Unknown, the failures have been happening sporadically in CI

### Expected behavior

Code should handle all input trace data, fall back gracefully to not returning CP if impossible to calculate.

### Relevant log output

_No response_

### Screenshot

_No response_

### Additional context

We need to rethink the [switch statement][1]. The relative positioning of parent/child spans has a fixed number of permutations, they should be covered by the appropriate if-conditions, ideally in a hierarchical manner, e.g. 
```
if child starts after parent
  if child ends before parent
    etc.
```

The function should also be simplified to do early returns. For example, instead of introducing a nesting level with
```
if (span && span.references.length) {
   // nested logic
}
```
replace it with
```
if (!(span && span.references.length)) {
   return
}
```

[1]: https://github.com/jaegertracing/jaeger-ui/blob/17eb2916047f7c3d1f5691957e29bbf6784b2e22/packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/sanitizeOverFlowingChildren.tsx#L36

### Jaeger backend version

_No response_

### SDK

_No response_

### Pipeline

_No response_

### Stogage backend

_No response_

### Operating system

_No response_

### Deployment model

_No response_

### Deployment configs

_No response_

### PR DESCRIPTION
**Pull request by GLVSKiriti**
|----parent----|
             |------------child------------|

The above case is not handled in previous code
Resolves #1779

### CODE REVIEW

File: packages/jaeger-ui/src/components/TracePage/CriticalPath/index.test.js [hunk 1]
```diff
@@ -22,22 +22,26 @@ import sanitizeOverFlowingChildren from './utils/sanitizeOverFlowingChildren';
...
+      const criticalPath = computeCriticalPath(sanitizedSpanMap, rootSpanId, []);
+      expect(criticalPath).toStrictEqual(testProps.criticalPathSections);
+    });
 
-  it('Critical path sections', () => {
-    const criticalPath = TraceCriticalPath(testProps.trace);
-    expect(criticalPath).toStrictEqual(testProps.criticalPathSections);
-  });
-});
+    it('Critical path sections', () => {
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/TracePage/CriticalPath/index.test.js:42]: these are meant to be read as a sentence, "it should find critical...". What do you test here?

**Review by yurishkuro — DISMISSED** [review]

File: packages/jaeger-ui/src/components/TracePage/CriticalPath/index.test.js (hunk 1 above)
Comment (GLVSKiriti) [reply packages/jaeger-ui/src/components/TracePage/CriticalPath/index.test.js:42]: Here I test whether the calculated critical path sections of a trace is same as expected critical path sections or not

File: packages/jaeger-ui/src/components/TracePage/CriticalPath/index.test.js [hunk 2]
```diff
@@ -22,22 +22,26 @@ import sanitizeOverFlowingChildren from './utils/sanitizeOverFlowingChildren';
...
-    }, new Map());
-    const refinedSpanMap = getChildOfSpans(spanMap);
-    const sanitizedSpanMap = sanitizeOverFlowingChildren(refinedSpanMap);
-    const criticalPath = computeCriticalPath(sanitizedSpanMap, rootSpanId, []);
-    expect(criticalPath).toStrictEqual(testProps.criticalPathSections);
-  });
+describe.each([[test1], [test2], [test3], [test4], [test5], [test6], [test7], [test8]])(
+  'Happy Path',
+  testProps => {
+    it('Should find criticalPathSections correctly', () => {
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/TracePage/CriticalPath/index.test.js:30]:
```suggestion
    it('should find criticalPathSections correctly', () => {
```

File: packages/jaeger-ui/src/components/TracePage/CriticalPath/index.test.js (hunk 1 above)
Comment (yurishkuro) [reply packages/jaeger-ui/src/components/TracePage/CriticalPath/index.test.js:42]: why do we even need two test functions here? The one above does a lot more work that is already done inside the business functions.

**Comment by GLVSKiriti** [comment]
Basically the first test is to test computeCriticalPath function.
And second test for the whole criticalPath function i.e, memoizedTraceCriticalPath

But memoizedTraceCriticalPath returns the same as computeCriticalPath function. Then should I remove the 2nd test case??

**Comment by yurishkuro** [comment]
I would remove the first one, it's calling internal implementation details which are irrelevant for the purpose of the test.

File: packages/jaeger-ui/src/components/TracePage/CriticalPath/index.test.js [hunk 3]
```diff
@@ -22,22 +22,14 @@ import sanitizeOverFlowingChildren from './utils/sanitizeOverFlowingChildren';
 import test6 from './testCases/test6';
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/TracePage/CriticalPath/index.test.js:22]: these imports are not needed?

Comment (GLVSKiriti) [reply packages/jaeger-ui/src/components/TracePage/CriticalPath/index.test.js:22]: But we want to test them right.  Without importing how can we do that?

Comment (yurishkuro) [reply packages/jaeger-ui/src/components/TracePage/CriticalPath/index.test.js:22]: not the tests, but the two implementation functions that you're no longer calling

Comment (yurishkuro) [reply packages/jaeger-ui/src/components/TracePage/CriticalPath/index.test.js:22]: I couldn't comment on the previous line

Comment (GLVSKiriti) [reply packages/jaeger-ui/src/components/TracePage/CriticalPath/index.test.js:22]: Okk I got it!

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**